src/Symfony/Component/Validator/Mapping/ClassMetadata.php

```diff
@@ -178,6 +180,14 @@ public function addConstraint(Constraint $constraint)
             throw new ConstraintDefinitionException(sprintf('The constraint "%s" cannot be put on classes.', \get_class($constraint)));
         }
 
+        if ($constraint instanceof Sequentially || $constraint instanceof Compound) {
```

Comment:
`All` could probably be added to this list, but could it lead to a BC break somehow?

Reply:
>All could probably be added to this list

Why? It does not allow classes already thanks to the `if` before 🤔

Reply:
Wrong path. I meant in `MemberMetadata` 😄

Reply:
Hmm, what about a check against `Composite` instead? It may be needed for both then, and even target 3.4?

Reply:
Alright, let's consider this a bug fix. I'll move the second commit to another PR for 3.4

Reply:
See #35843